### PR TITLE
ci(NODE-4651): add Node.js 18

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR adds Node18 to our CI matrix.

##### Is there new documentation needed for these changes?

No.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/integration/client-side-encryption/client_side_encryption.spec.test.ts [hunk 1]
```diff
@@ -80,5 +81,8 @@ describe('Client Side Encryption (Legacy)', function () {
 });
 
 describe('Client Side Encryption (Unified)', function () {
+  if (process.version.includes('18')) {
+    setDefaultResultOrder('ipv4first');
+  }
```
Comment (nbbeeken) [inline test/integration/client-side-encryption/client_side_encryption.spec.test.ts:86]: Should this be in a before and then reset inside an after?

Comment (baileympearson) [reply test/integration/client-side-encryption/client_side_encryption.spec.test.ts:86]: sure, done

File: test/integration/client-side-encryption/client_side_encryption.spec.test.ts [hunk 2]
```diff
@@ -81,8 +81,16 @@ describe('Client Side Encryption (Legacy)', function () {
 });
 
 describe('Client Side Encryption (Unified)', function () {
-  if (process.version.includes('18')) {
-    setDefaultResultOrder('ipv4first');
-  }
+  before(function () {
+    if (process.version.includes('18')) {
```
Comment (nbbeeken) [inline test/integration/client-side-encryption/client_side_encryption.spec.test.ts:98]: 16.17 is the current minor so this is going to run on 16.18 when it comes out, let's make sure this is running for the major 18 only

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: test/tools/runner/hooks/configuration.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-'use strict';
+('use strict');
```
Comment (nbbeeken) [inline test/tools/runner/hooks/configuration.js:1]: strange formatting, intentional?

**Review by nbbeeken — CHANGES_REQUESTED** [review]

**Review by nbbeeken — APPROVED** [review]

**Review by durran — APPROVED** [review]

**Pull request merged**